File: scala/src/main/scala/SumOfMultiples.scala
object SumOfMultiples {
  def multiples(factor: Int, limit: Int): Set[Int] = {
    (factor until limit by factor).toSet
  }

  def sum(factors: Set[Int], limit: Int): Int = {
    factors.flatMap(multiples(_, limit)).sum
  }
}

File: python/src/sum_of_multiples.py
def sum_of_multiples(limit, multiples):
    unique_multiples = set()
    for multiple in multiples:
        unique_multiples = unique_multiples.union(get_multiples(limit, multiple))
    return sum(unique_multiples)

def get_multiples(limit, multiple):
    if multiple == 0:
        return {0}
    return {s for s in range(multiple, limit, multiple)}
